### PR TITLE
WT-7005 Write Session subpage for Architecture Guide

### PR DESCRIPTION
**Pull request by etienneptl**

### CODE REVIEW

File: src/docs/arch-session.dox [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+application can start sending requests to WiredTiger using a session. A session is internally
+represented by WT_SESSION and plays an important role since almost all operations require one.
+
+A session can only be created through an existing connection with the API
+WT_CONNECTION::open_session. It is possible to create multiple sessions through the same connection.
+In fact, one connection can have multiple sessions but one session can only be associated to one
+connection.
+
+@section arch_session_ops Operations
+The different operations a session can request to WiredTiger are related to cursors handles, table
```
Comment (quchenhao) [inline src/docs/arch-session.dox:14]: Within a session, user can open various types of cursors to perform specified operations?

File: src/docs/arch-session.dox [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+operations and transactions. You can read the complete description of each possible operation in the
+documentation related to WT_SESSION.
+
+@section arch_session_txn Transactions
+It is possible to group a number of operations within a session and get a single result, in other
```
Comment (quchenhao) [inline src/docs/arch-session.dox:19]: "get a single result" is a bit strange. I would simply say group them in a transaction.

File: src/docs/arch-session.dox [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+words, multiple operations can be treated as a single atomic operation. This can be done using @ref
+arch-transaction. When multiple operations are part of a transaction, they must either all be
+successful or fail; WiredTiger will cancel any successful operation before a failure within the same
```
Comment (quchenhao) [inline src/docs/arch-session.dox:22]: cancel -> abort? "successful" is confusing.

File: src/docs/arch-session.dox [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+transaction. Finally, one session can hold a unique transaction at any given time and this
+transaction only belongs to that same session.
+
+@section arch_session_cur Cursors
+Session can perform operations on multiple collections using cursors (see @ref arch-cursor for more
+details). Indeed, a session can have multiple cursors of any types and they all share the same
+transaction that is associated to the session. It is also possible to cache those cursors if
```
Comment (quchenhao) [inline src/docs/arch-session.dox:29]: transaction context?

File: src/docs/arch-session.dox [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+required through the configuration string given to WT_CONNECTION::open_session or ::wiredtiger_open.
+
+@section arch_session_dhandles Data Handles
+During its lifetime, a session can accumulate a list of data handles (see @ref arch-dhandle). A
+session acquires a data handle every time is needs to perform an operation on a table and the data
```
Comment (quchenhao) [inline src/docs/arch-session.dox:34]: every time it needs

File: src/docs/arch-session.dox [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+handle is not already part of the list. Once a session no longer needs to operate on a table, it
+will mark the associated data handle as idle. This helps the sweep server to release data handles
+that are inactive, see @ref arch-dhandle-lifecycle for more details.
+
+@section arch_session_closure Closure
+When a connection is no longer needed, it can be closed using WT_SESSION::close. This closes the
+transaction and the cursors associated to the session and frees all the resources that are exclusive
+to the session.
```
Comment (quchenhao) [inline src/docs/arch-session.dox:42]: I think we should mention that session is always executed as a single thread. We can run multiple sessions in a connection. A single thread can start multiple sessions.

File: src/docs/arch-session.dox [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
```
Comment (quchenhao) [inline src/docs/arch-session.dox:43]: We should also the maximum concurrent sessions is configured at connection start and it is fixed across connection.

Comment (quchenhao) [reply src/docs/arch-session.dox:43]: May also mention internal sessions and user sessions. What are the differences and how are the internal sessions used within WiredTiger.

Comment (quchenhao) [reply src/docs/arch-session.dox:43]: You can set isolation level using session config.

Comment (etienneptl) [reply src/docs/arch-session.dox:43]: Thank you for those great suggestions

File: src/docs/arch-session.dox (hunk 1 above)
Comment (etienneptl) [reply src/docs/arch-session.dox:14]: I already say a session can have various cursors in the Cursor section. I took the headers from the documentation [here](https://source.wiredtiger.com/develop/struct_w_t___s_e_s_s_i_o_n.html#details). I will see if I can reword differently.

File: src/docs/arch-session.dox (hunk 2 above)
Comment (etienneptl) [reply src/docs/arch-session.dox]: See 339e6d00b48db9f60bd28f04c4046825929dcb6a

File: src/docs/arch-session.dox (hunk 3 above)
Comment (etienneptl) [reply src/docs/arch-session.dox]: I changed `cancel` to `abort` in 339e6d00b48db9f60bd28f04c4046825929dcb6a.
I left successful as I think it sounds correct. Can we have more thoughts about it ? @bvpvamsikrishna @AlexeyAnisimovAU

File: src/docs/arch-session.dox (hunk 4 above)
Comment (etienneptl) [reply src/docs/arch-session.dox]: See 339e6d00b48db9f60bd28f04c4046825929dcb6a

File: src/docs/arch-session.dox (hunk 5 above)
Comment (etienneptl) [reply src/docs/arch-session.dox]: Thanks ! Fixed in 339e6d00b48db9f60bd28f04c4046825929dcb6a

File: src/docs/arch-session.dox (hunk 6 above)
Comment (etienneptl) [reply src/docs/arch-session.dox:42]: I will add a multithreading section that will refer to this [page](https://source.wiredtiger.com/develop/threads.html).

File: src/docs/arch-session.dox (hunk 7 above)
Comment (etienneptl) [reply src/docs/arch-session.dox:44]: > We should also mention the maximum concurrent sessions is configured at connection start and it is fixed across connection.
See b9cda0dec96a5059e85a8344ba21723a51ca3191

Comment (etienneptl) [reply src/docs/arch-session.dox:49]: > May be also worth mentioning internal sessions and user sessions. What are the differences and how are the internal sessions used within WiredTiger.

See ba2107c705994f0ccd4c6c0e1ca78a4558244351. I gave a rough definition, were you thinking of more  ?

File: src/docs/arch-session.dox (hunk 6 above)
Comment (etienneptl) [reply src/docs/arch-session.dox:48]: See 835ad1ce1878ebaa49f624738a0336bd35d3328a

File: src/docs/arch-session.dox [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+application can start sending requests to WiredTiger using a session. A session is internally
+represented by WT_SESSION and plays an important role since almost all operations require one.
+
+A session can only be created through an existing connection with the API
+WT_CONNECTION::open_session. It is possible to create multiple sessions through the same connection.
+In fact, one connection can have multiple sessions but one session can only be associated to one
+connection. The maximum number of sessions is set through the configuration item \c session_max as
+part of the configuration string given in ::wiredtiger_open.
+
+Sessions that are created by the calling application are called "user session". It exists another
```
Comment (AlexeyAnisimovAU) [inline src/docs/arch-session.dox:14]: Maybe replace `It exists` with `There is`

File: src/docs/arch-session.dox [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+type of session called "internal session". Internal sessions are created and managed by WiredTiger
+internally and they are associated to global services at runtime such as @ref arch-eviction, @ref
```
Comment (AlexeyAnisimovAU) [inline src/docs/arch-session.dox:16]: I believe it should be either `associated with` or `related to`.

File: src/docs/arch-session.dox [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+represented by WT_SESSION and plays an important role since almost all operations require one.
+
+A session can only be created through an existing connection with the API
+WT_CONNECTION::open_session. It is possible to create multiple sessions through the same connection.
+In fact, one connection can have multiple sessions but one session can only be associated to one
+connection. The maximum number of sessions is set through the configuration item \c session_max as
+part of the configuration string given in ::wiredtiger_open.
+
+Sessions that are created by the calling application are called "user session". It exists another
+type of session called "internal session". Internal sessions are created and managed by WiredTiger
```
Comment (AlexeyAnisimovAU) [inline src/docs/arch-session.dox:15]: I feel like we need to rephrase this sentence. The previous sentence ends with "internal session". This one starts with the same words and in the middle again "WiredTiger internally". Maybe we can start this sentence with `This type of sessions are created...`?

File: src/docs/arch-session.dox [hunk 11]
```diff
@@ -0,0 +1,53 @@
+/*! @arch_page arch-session Session
+
+@section arch_session_def Definition
+After a @ref arch-connection has been established between the application and WiredTiger, the
+application can start sending requests to WiredTiger using a session. A session is internally
+represented by WT_SESSION and plays an important role since almost all operations require one.
+
+A session can only be created through an existing connection with the API
+WT_CONNECTION::open_session. It is possible to create multiple sessions through the same connection.
+In fact, one connection can have multiple sessions but one session can only be associated to one
```
Comment (AlexeyAnisimovAU) [inline src/docs/arch-session.dox:10]: I believe `associated to` is not grammatically correct. It should be either `associated with` or `related to`.

File: src/docs/arch-session.dox (hunk 8 above)
Comment (AlexeyAnisimovAU) [reply src/docs/arch-session.dox:14]: It should be `are called "user sessions"`, plural, not singular.

File: src/docs/arch-session.dox [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+part of the configuration string given in ::wiredtiger_open.
+
+Sessions that are created by the calling application are called "user session". It exists another
+type of session called "internal session". Internal sessions are created and managed by WiredTiger
+internally and they are associated to global services at runtime such as @ref arch-eviction, @ref
+arch-checkpoint, @ref arch-rts, ...
+
+@section arch_session_ops Operations
+The different operations a session can request to WiredTiger are related to cursors handles, table
+operations and transactions. You can read the complete description of each possible operation in the
```
Comment (AlexeyAnisimovAU) [inline src/docs/arch-session.dox:21]: Missing comma before `and`.

File: src/docs/arch-session.dox [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+It is possible to group a number of operations within a session, in other words, multiple operations
+can be treated as a single atomic operation. This can be done using @ref arch-transaction. When
+multiple operations are part of a transaction, they must either all be successful or fail;
+WiredTiger will abort any successful operation before a failure within the same transaction.
+Finally, one session can hold a unique transaction at any given time and this transaction only
+belongs to that same session.
+
+@section arch_session_cur Cursors
+Session can perform operations on multiple collections using cursors (see @ref arch-cursor for more
+details). Indeed, a session can have multiple cursors of any types and they all share the same
```
Comment (AlexeyAnisimovAU) [inline src/docs/arch-session.dox:34]: `any type`

File: src/docs/arch-session.dox [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+transaction context that is associated to the session. It is also possible to cache those cursors if
```
Comment (AlexeyAnisimovAU) [inline src/docs/arch-session.dox:35]: `associated with` or `related to`

File: src/docs/arch-session.dox [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+@section arch_session_dhandles Data Handles
+During its lifetime, a session can accumulate a list of data handles (see @ref arch-dhandle). A
+session acquires a data handle every time it needs to perform an operation on a table and the data
+handle is not already part of the list. Once a session no longer needs to operate on a table, it
+will mark the associated data handle as idle. This helps the sweep server to release data handles
+that are inactive, see @ref arch-dhandle-lifecycle for more details.
+
+@section arch_session_closure Closure
+When a connection is no longer needed, it can be closed using WT_SESSION::close. This closes the
+transaction and the cursors associated to the session and frees all the resources that are exclusive
```
Comment (AlexeyAnisimovAU) [inline src/docs/arch-session.dox:47]: Same here, `associated with` or `related to`

File: src/docs/arch-session.dox [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+operations and transactions. You can read the complete description of each possible operation in the
+documentation related to WT_SESSION.
+
+@section arch_session_txn Transactions
+It is possible to group a number of operations within a session, in other words, multiple operations
+can be treated as a single atomic operation. This can be done using @ref arch-transaction. When
+multiple operations are part of a transaction, they must either all be successful or fail;
+WiredTiger will abort any successful operation before a failure within the same transaction.
+Finally, one session can hold a unique transaction at any given time and this transaction only
+belongs to that same session.
```
Comment (AlexeyAnisimovAU) [inline src/docs/arch-session.dox:30]: Can a session have multiple running transactions?

File: src/docs/arch-session.dox [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+Session can perform operations on multiple collections using cursors (see @ref arch-cursor for more
+details). Indeed, a session can have multiple cursors of any types and they all share the same
+transaction context that is associated to the session. It is also possible to cache those cursors if
+required through the configuration string given to WT_CONNECTION::open_session or ::wiredtiger_open.
+
+@section arch_session_dhandles Data Handles
+During its lifetime, a session can accumulate a list of data handles (see @ref arch-dhandle). A
+session acquires a data handle every time it needs to perform an operation on a table and the data
+handle is not already part of the list. Once a session no longer needs to operate on a table, it
+will mark the associated data handle as idle. This helps the sweep server to release data handles
```
Comment (AlexeyAnisimovAU) [inline src/docs/arch-session.dox:42]: Maybe it will sound more natural if we swap the first and the second sentences?

File: src/docs/arch-session.dox [hunk 18]
```diff
@@ -0,0 +1,53 @@
...
+
+@section arch_session_closure Closure
+When a connection is no longer needed, it can be closed using WT_SESSION::close. This closes the
+transaction and the cursors associated to the session and frees all the resources that are exclusive
+to the session.
+
+@section arch_session_thread Multithreading
+A session is always executed as a single thread, see @ref threads for more details.
+
+*/
```
Comment (AlexeyAnisimovAU) [inline src/docs/arch-session.dox:53]: What happens to inactive sessions? I heard they get cleared over time?

File: src/docs/arch-session.dox [hunk 19]
```diff
@@ -0,0 +1,53 @@
...
+@section arch_session_txn Transactions
+It is possible to group a number of operations within a session, in other words, multiple operations
+can be treated as a single atomic operation. This can be done using @ref arch-transaction. When
+multiple operations are part of a transaction, they must either all be successful or fail;
+WiredTiger will abort any successful operation before a failure within the same transaction.
+Finally, one session can hold a unique transaction at any given time and this transaction only
+belongs to that same session.
+
+@section arch_session_cur Cursors
+Session can perform operations on multiple collections using cursors (see @ref arch-cursor for more
```
Comment (bvpvamsikrishna) [inline src/docs/arch-session.dox:33]: How about rewording this as "A session can perform multiple data operations on same/different collections using multiple cursors (see @ref arch-cursor for more details). All the cursors associated with a session will share that session transaction context.

File: src/docs/arch-session.dox [hunk 20]
```diff
@@ -0,0 +1,53 @@
...
+details). Indeed, a session can have multiple cursors of any types and they all share the same
+transaction context that is associated to the session. It is also possible to cache those cursors if
+required through the configuration string given to WT_CONNECTION::open_session or ::wiredtiger_open.
+
+@section arch_session_dhandles Data Handles
+During its lifetime, a session can accumulate a list of data handles (see @ref arch-dhandle). A
+session acquires a data handle every time it needs to perform an operation on a table and the data
```
Comment (bvpvamsikrishna) [inline src/docs/arch-session.dox:40]: When a session access a table for the first time, the data handle of that table is acquired and cached.

File: src/docs/arch-session.dox [hunk 21]
```diff
@@ -0,0 +1,53 @@
...
+handle is not already part of the list. Once a session no longer needs to operate on a table, it
+will mark the associated data handle as idle. This helps the sweep server to release data handles
+that are inactive, see @ref arch-dhandle-lifecycle for more details.
+
+@section arch_session_closure Closure
+When a connection is no longer needed, it can be closed using WT_SESSION::close. This closes the
```
Comment (bvpvamsikrishna) [inline src/docs/arch-session.dox:46]: typo... isn't it is session instead of connection?

File: src/docs/arch-session.dox (hunk 16 above)
Comment (etienneptl) [reply src/docs/arch-session.dox:30]: No @AlexeyAnisimovAU, it cannot.

File: src/docs/arch-session.dox (hunk 18 above)
Comment (etienneptl) [reply src/docs/arch-session.dox:53]: That is a good point, I will add a few words about it.

File: src/docs/arch-session.dox (hunk 8 above)
Comment (etienneptl) [reply src/docs/arch-session.dox:14]: It is much clearer this way @bvpvamsikrishna , thanks!

File: src/docs/arch-session.dox (hunk 11 above)
Comment (etienneptl) [reply src/docs/arch-session.dox]: See b5dec2eef112941c25aefba09a01e2b0e572cc17

File: src/docs/arch-session.dox (hunk 9 above)
Comment (etienneptl) [reply src/docs/arch-session.dox]: See b5dec2eef112941c25aefba09a01e2b0e572cc17

File: src/docs/arch-session.dox (hunk 14 above)
Comment (etienneptl) [reply src/docs/arch-session.dox]: See b5dec2eef112941c25aefba09a01e2b0e572cc17

File: src/docs/arch-session.dox (hunk 15 above)
Comment (etienneptl) [reply src/docs/arch-session.dox]: See b5dec2eef112941c25aefba09a01e2b0e572cc17

File: src/docs/arch-session.dox [hunk 22]
```diff
@@ -0,0 +1,53 @@
...
+arch-checkpoint, @ref arch-rts, ...
+
+@section arch_session_ops Operations
+The different operations a session can request to WiredTiger are related to cursors handles, table
+operations and transactions. You can read the complete description of each possible operation in the
+documentation related to WT_SESSION.
+
+@section arch_session_txn Transactions
+It is possible to group a number of operations within a session, in other words, multiple operations
+can be treated as a single atomic operation. This can be done using @ref arch-transaction. When
```
[the remark this answers is not in the document]
Comment (etienneptl) [reply src/docs/arch-session.dox]: Removed in 4ed74482ea0d3efb778724b7063098f1531af700

File: src/docs/arch-session.dox (hunk 16 above)
[the remark this answers is not in the document]
Comment (etienneptl) [reply src/docs/arch-session.dox]: Removed in 3964d517eb8cdce597244ee94edd344752457356

File: src/docs/arch-session.dox (hunk 13 above)
Comment (etienneptl) [reply src/docs/arch-session.dox]: Fixed in 4d91ebde292489e965f32a661fcb02d6f65084dc

File: src/docs/arch-session.dox (hunk 19 above)
Comment (etienneptl) [reply src/docs/arch-session.dox]: I like your version, used in 7082c5cf42133a5c0d7b223c9c30c3fdf41cefa9

File: src/docs/arch-session.dox (hunk 21 above)
Comment (etienneptl) [reply src/docs/arch-session.dox]: Correct.. ! 8689a6420c4a332bd19fc78affc6d6e64af0c1aa

File: src/docs/arch-session.dox (hunk 20 above)
Comment (etienneptl) [reply src/docs/arch-session.dox]: Reworded in 8689a6420c4a332bd19fc78affc6d6e64af0c1aa

File: src/docs/arch-session.dox (hunk 12 above)
Comment (etienneptl) [reply src/docs/arch-session.dox]: Added in 5130d5e96be27d801f62a20f5d3e81d0df110a77

File: src/docs/arch-session.dox (hunk 8 above)
Comment (etienneptl) [reply src/docs/arch-session.dox]: Edited in 40e60757a00fe080283aaa542965ce88620cb32b

File: src/docs/arch-session.dox (hunk 10 above)
Comment (etienneptl) [reply src/docs/arch-session.dox]: @AlexeyAnisimovAU  The section has been reworked in 40e6075, the word "internal" is still used a few times but it does not disturb me that much. What do you think ?

File: src/docs/arch-session.dox (hunk 1 above)
Comment (etienneptl) [reply src/docs/arch-session.dox]: Thanks for the suggestion @bvpvamsikrishna, see 7f22366dbbf139351ee9fcef71092098804fd14e
In the very first paragraph of this page, I wrote `the application can start sending requests to WiredTiger using a session`, is it confusing too since I say `can start sending requests` as well ?

File: src/docs/arch-session.dox (hunk 7 above)
Comment (etienneptl) [reply src/docs/arch-session.dox:47]: > You can set isolation level using session config.

Since it is a transaction concept, it should be detailed in the transaction page.

File: src/docs/arch-session.dox (hunk 17 above)
Comment (etienneptl) [reply src/docs/arch-session.dox]: It has been reworked as per `8689a64`, what do you think ?

File: src/docs/arch-session.dox (hunk 18 above)
Comment (etienneptl) [reply src/docs/arch-session.dox:52]: Added as part of 4724bec03a6ad8013c7ac390b9ba80f5dce99f7d

File: src/docs/arch-session.dox [hunk 23]
```diff
@@ -0,0 +1,51 @@
+/*! @arch_page arch-session Session
+
+@section arch_session_def Definition
+After a @ref arch-connection has been established between the application and WiredTiger, the
+application can start sending requests to WiredTiger using a session. A session is internally
+represented by WT_SESSION and plays an important role since almost all operations require one.
```
Comment (quchenhao) [inline src/docs/arch-session.dox:6]: almost all operations are performed under the context of a session?

File: src/docs/arch-session.dox [hunk 24]
```diff
@@ -0,0 +1,51 @@
...
+associated with one connection. The maximum number of sessions is set through the configuration item
+\c session_max as part of the configuration string in ::wiredtiger_open.
+
+Sessions that are created by the calling application are called "user sessions". WiredTiger also
+performs some internal operations such as @ref arch-eviction through self-created sessions. These
+sessions are called "internal sessions". The usage rules and guidelines for both internal sessions
+and user sessions are the same and the only difference between them is their origin of creation.
+
+@section arch_session_ops Operations
+The different operations that can be performed on a WiredTiger session are related to cursors
```
Comment (quchenhao) [inline src/docs/arch-session.dox:20]: cursors'

Comment (quchenhao) [reply src/docs/arch-session.dox:20]: What is cursors handles

File: src/docs/arch-session.dox [hunk 25]
```diff
@@ -0,0 +1,51 @@
...
+@section arch_session_cur Cursors
+A session can perform multiple data operations on one or several collections using multiple cursors
+(see @ref arch-cursor for more details). All the cursors associated with a session share that
+session transaction context. It is also possible to cache those cursors if required through the
+configuration string given to WT_CONNECTION::open_session or ::wiredtiger_open.
+
+@section arch_session_dhandles Data Handles
+During its lifetime, a session can accumulate a list of data handles (see @ref arch-dhandle).
+Indeed, when a session accesses a table for the first time, the data handle of that table is
+acquired and cached. Once a session no longer needs to operate on a table, it marks the associated
```
Comment (quchenhao) [inline src/docs/arch-session.dox:39]: I would say file instead of table here.

File: src/docs/arch-session.dox (hunk 24 above)
Comment (etienneptl) [reply src/docs/arch-session.dox:20]: See the section dedicated to cursor handles : https://source.wiredtiger.com/develop/struct_w_t___s_e_s_s_i_o_n.html
The API is open_cursor

File: src/docs/arch-session.dox (hunk 25 above)
Comment (etienneptl) [reply src/docs/arch-session.dox:39]: Are you saying you would replace "table" with "file" for the three occurrences lines 38/39 ?

File: src/docs/arch-session.dox (hunk 23 above)
Comment (etienneptl) [reply src/docs/arch-session.dox]: See b9544afeedb411abcfede0d56b7eea041b152ec7

File: src/docs/arch-session.dox (hunk 24 above)
Comment (quchenhao) [reply src/docs/arch-session.dox:21]: Yes, you are talking about the schema operations. I wouldn't call then handles as it is confusing with dhandles.

File: src/docs/arch-session.dox (hunk 25 above)
Comment (quchenhao) [reply src/docs/arch-session.dox:40]: yes, in the data handle section as dhandle is associated with a file not a table.

File: src/docs/arch-session.dox [hunk 26]
```diff
@@ -0,0 +1,52 @@
...
+data handle as idle. This helps the sweep server release data handles that are inactive, see @ref
+arch-dhandle-lifecycle for more details.
+
+@section arch_session_closure Closure
+A session can be closed using WT_CONNECTION::close. Closing the connection will also close all
+opened sessions. When a session is closed, it releases all the resources associated with the session
+handle, including rolling back any active transaction and closing the cursors that are still open.
```
Comment (quchenhao) [inline src/docs/arch-session.dox:47]: What is a session handle? Can we avoid the term handle?

File: src/docs/arch-session.dox (hunk 18 above)
Comment (quchenhao) [reply src/docs/arch-session.dox:52]: Inactive dhandles are swept away but I don't think we clear inactive sessions.

File: src/docs/arch-session.dox [hunk 27]
```diff
@@ -0,0 +1,52 @@
...
+It is possible to group several operations within a session, in other words, multiple operations can
+be treated as a single atomic operation. This can be done using @ref arch-transaction. Furthermore,
+one session can hold a unique transaction at any given time and this transaction only belongs to
+that session.
+
+@section arch_session_cur Cursors
+A session can perform multiple data operations on one or several collections using multiple cursors
+(see @ref arch-cursor for more details). All the cursors associated with a session share that
+session transaction context. It is also possible to cache those cursors if required through the
+configuration string given to WT_CONNECTION::open_session or ::wiredtiger_open.
```
Comment (AlexeyAnisimovAU) [inline src/docs/arch-session.dox:35]: Since you mentioned cached cursors and config string, maybe we should provide the config option here? The same as you did with `session_max`.

File: src/docs/arch-session.dox (hunk 24 above)
Comment (etienneptl) [reply src/docs/arch-session.dox:21]: See 14e531c4083f3864a614c7812c433a8f5227914d

File: src/docs/arch-session.dox (hunk 26 above)
Comment (etienneptl) [reply src/docs/arch-session.dox:47]: I agree it could be confusing with the other usage of data handles.

Comment (etienneptl) [reply src/docs/arch-session.dox]: Fixed in bf2d6631d16c0686b62484308fbaaf718db99865

File: src/docs/arch-session.dox (hunk 27 above)
Comment (etienneptl) [reply src/docs/arch-session.dox:35]: Sounds like a good idea !

Comment (etienneptl) [reply src/docs/arch-session.dox]: fdcceeeef202c196459e750569e8288dc766ca49

File: src/docs/arch-session.dox (hunk 21 above)
Comment (etienneptl) [reply src/docs/arch-session.dox]: In fact, since I reworded the section, I had to use `WT_SESSION` here, see 659c5aaf197846c9ecd04ab823bd1e939002ba42

File: src/docs/arch-session.dox (hunk 16 above)
Comment (AlexeyAnisimovAU) [reply src/docs/arch-session.dox]: Maybe we should mention that here?

Comment (etienneptl) [reply src/docs/arch-session.dox]: I think it is clear with the following sentence:
`Furthermore,
one session can hold a unique transaction at any given time and this transaction only belongs to
that session.`

Comment (AlexeyAnisimovAU) [reply src/docs/arch-session.dox]: To me, `unique` doesn't imply that a session can have _only one_ running transaction. Maybe if we replace 'a unique' with `only one active` it would sound more clear? Something like this: 
`Furthermore, a session can hold only one running transaction at any given time and this transaction only belongs to that session.`
We need to make sure the reader understands that there's a one-to-one relationship.

Comment (etienneptl) [reply src/docs/arch-session.dox]: I will make the change, it is effectively clearer this way. Thanks !

Comment (etienneptl) [reply src/docs/arch-session.dox]: See 96500e52cc2c6e42d2507892f24a6b0c9b955371

File: src/docs/arch-session.dox [hunk 28]
```diff
@@ -0,0 +1,53 @@
...
+represented by WT_SESSION and plays an important role since almost all operations are performed
+under the context of a session.
+
+A session can only be created through an existing connection with the API
+WT_CONNECTION::open_session and it is possible to create multiple sessions through the same
+connection. In fact, one connection can have multiple sessions but one session can only be
+associated with one connection. The maximum number of sessions is set through the configuration item
+\c session_max as part of the configuration string in ::wiredtiger_open.
+
+Sessions that are created by the calling application are called "user sessions". WiredTiger also
```
Comment (AlexeyAnisimovAU) [inline src/docs/arch-session.dox:15]: I think we can remove `that are`. It would be `Sessions created by`.

Comment (etienneptl) [reply src/docs/arch-session.dox]: Thanks, see 3d33e489c59c6a7e0a528b6dd418bc2b8bc1161b

**Review by AlexeyAnisimovAU — APPROVED** [review]
LGTM

File: src/docs/arch-session.dox [hunk 29]
```diff
@@ -0,0 +1,53 @@
...
+associated with one connection. The maximum number of sessions is set through the configuration item
+\c session_max as part of the configuration string in ::wiredtiger_open.
+
+Sessions created by the calling application are called "user sessions". WiredTiger also performs
+some internal operations such as @ref arch-eviction through self-created sessions. These sessions
+are called "internal sessions". The usage rules and guidelines for both internal sessions and user
+sessions are the same and the only difference between them is their origin of creation.
+
+@section arch_session_ops Operations
+The different operations that can be performed on a WiredTiger session are related to cursors
```
Comment (quchenhao) [inline src/docs/arch-session.dox:21]: cursors‘ creation？

**Review by quchenhao — APPROVED** [review]
lgtm with a small comment

File: src/docs/arch-session.dox (hunk 29 above)
Comment (etienneptl) [reply src/docs/arch-session.dox]: Reworded in c53b1a106d811f2489ff003eff3430e42fedebfb

**Review by bvpvamsikrishna — APPROVED** [review]
lgtm
